manticore/core/smtlib/visitors.py

```diff
@@ -548,19 +544,20 @@ def arithmetic_simplify(expression):
 
 
 def to_constant(expression):
-    value = arithmetic_simplify(expression)
+    value = simplify(expression)
     if isinstance(value, Constant):
         return value.value
     elif isinstance(value, Array):
-        if value.index_max:
+        if expression.index_max:
```

Comment:
This should be `value`. And `simplify()` should maintain the index_max pre/pos simplification